server/src/internalClusterTest/java/org/opensearch/search/profile/aggregation/AggregationProfilerIT.java

```diff
@@ -843,13 +875,39 @@ public void testGlobalAggWithStatsSubAggregatorProfile() {
             if (globalAggResult.getMaxSliceTime() != null) {
                 // concurrent segment search enabled
                 assertEquals(CONCURRENT_SEARCH_BREAKDOWN_KEYS, breakdown.keySet());
+                for (QueryProfileShardResult collectorResult : profileShardResult.getQueryProfileResults()) {
+                    assertCollectorResultWithConcurrentSearchEnabled(collectorResult, 0);
+                }
             } else {
                 assertEquals(BREAKDOWN_KEYS, breakdown.keySet());
+                for (QueryProfileShardResult collectorResult : profileShardResult.getQueryProfileResults()) {
+                    assertCollectorResult(collectorResult, 0);
+                }
             }
             assertThat(breakdown.get(INITIALIZE), greaterThan(0L));
             assertThat(breakdown.get(COLLECT), greaterThan(0L));
             assertThat(breakdown.get(BUILD_AGGREGATION).longValue(), greaterThan(0L));
             assertEquals(0, breakdown.get(REDUCE).intValue());
         }
     }
+
+    private void assertCollectorResult(QueryProfileShardResult collectorResult, int expectedChildrenCount) {
+        long nodeTime = collectorResult.getCollectorResult().getTime();
+        assertThat(collectorResult.getCollectorResult().getMaxSliceTime(), equalTo(nodeTime));
+        assertThat(collectorResult.getCollectorResult().getMinSliceTime(), equalTo(nodeTime));
+        assertThat(collectorResult.getCollectorResult().getAvgSliceTime(), equalTo(nodeTime));
+        assertThat(collectorResult.getCollectorResult().getReduceTime(), equalTo(0L));
+        assertThat(collectorResult.getCollectorResult().getSliceCount(), equalTo(1));
+        assertThat(collectorResult.getCollectorResult().getProfiledChildren().size(), equalTo(expectedChildrenCount));
+    }
+
+    private void assertCollectorResultWithConcurrentSearchEnabled(QueryProfileShardResult collectorResult, int expectedChildrenCount) {
+        long nodeTime = collectorResult.getCollectorResult().getTime();
+        assertThat(collectorResult.getCollectorResult().getMaxSliceTime(), lessThanOrEqualTo(nodeTime));
+        assertThat(collectorResult.getCollectorResult().getMinSliceTime(), lessThanOrEqualTo(nodeTime));
+        assertThat(collectorResult.getCollectorResult().getAvgSliceTime(), lessThanOrEqualTo(nodeTime));
+        assertThat(collectorResult.getCollectorResult().getReduceTime(), greaterThan(0L));
+        assertThat(collectorResult.getCollectorResult().getSliceCount(), greaterThanOrEqualTo(1));
+        assertThat(collectorResult.getCollectorResult().getProfiledChildren().size(), equalTo(expectedChildrenCount));
```

Comment:
Lets add an assert to verify that atleast one children `reason` is `aggregation`. Same for method `assertCollectorResult`

Reply:
Added now.